airflow/operators/trigger_dagrun.py

```diff
@@ -40,9 +42,27 @@
 
 
 if TYPE_CHECKING:
+    from sqlalchemy.orm.session import Session
+
     from airflow.models.taskinstance import TaskInstanceKey
 
 
+def _parse_execution_date(date: datetime.datetime | str | None) -> datetime.datetime:
+    if isinstance(date, datetime.datetime):
+        return date
+    elif isinstance(date, str):
+        return timezone.parse(date)
+    else:
+        return timezone.utcnow()
```

Comment:
This else block feels unintuitive from the function name. Either the name should be more specific (giving context why defaulting to now is correct), or raising a ValueError is better to put the default logic at the call site.